src/util/createServerRootMixin.js

```diff
@@ -184,30 +155,23 @@ function augmentInstantSearch(
    * @returns {void}
    */
   search.__forceRender = function(widget, parent) {
-    const localHelper = parent.getHelper();
-
-    const results = search.__initialSearchResults[parent.getIndexId()];
-
     // this happens when a different InstantSearch gets rendered initially,
     // after the hydrate finished. There's thus no initial results available.
-    if (!results) {
+    if (parent.getResults() === null) {
       return;
     }
-
+    const results = parent.getResults();
```

Comment:
We can declare this variable above to also use it in the condition.

Reply:
6e69caf